AEPMessaging/Sources/MessagingPropositionItem+Tracking.swift

```diff
@@ -0,0 +1,92 @@
+/*
+ Copyright 2023 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import AEPCore
+import AEPServices
+
+@objc
+public extension MessagingPropositionItem {
+    /// Tracks interaction with the given proposition item.
+    ///
+    /// - Parameter interaction: an enum specifying the type of interactioon.
+    func track(interaction: MessagingEdgeEventType) {
```

Comment:
We are sending a tracking event for IAM here https://github.com/adobe/aepsdk-messaging-ios/blob/da23275cd7891f6f304670c1cd5dee8d6f421768/AEPMessaging/Sources/Messaging%2BEdgeEvents.swift#L287

How do you feel about moving that case also here and sending appropriate tracking details based on the schema type?

Reply:
Yeah, missed stating it in the pr. I plan to push an update for merging the two workflows & adding item level support.